src/contexts/PermissionsContext.js

```diff
@@ -0,0 +1,125 @@
+import React from 'react'
+import {
+  appRoles,
+  entityResolver,
+  roleResolver,
+  entityRoles,
+  permissionsByEntity,
+} from '../permissions'
+import { objectFromEntries } from '../utils'
+
+const { Provider, Consumer } = React.createContext()
+
+// props that are directly converted to state
+const PROPS_TO_STATE = ['apps', 'permissions', 'wrapper']
```

Comment:
Do we need to convert these all to state? It seems like we could get away without it, and just merge props and state when the provider's rendered?

Reply:
Yes you’re right!